### PR TITLE
WIP: POC with using ertclient data model

### PR DESCRIPTION
**Pull request by xjules**
POC for data being served via `ertviz.ertapi.ensemble`; open for discussion. 
TODO:
- Selection from realizations works, from parameter space to realizations is not implemented yet.
- Data caching
- Add tests and fix existing ones

### CODE REVIEW

File: ertviz/ertapi/ensemble/ensemble.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from ertviz.ertapi.ensemble.request_data import RequestData
+
+
+class Ensemble(RequestData):
+    def __init__(self, request_handler, metadata_dict):
```
Comment (ManInFez) [inline ertviz/ertapi/ensemble/ensemble.py:5]: metadata_dict should be renamed metadata_url I suppose?

File: ertviz/ertapi/ensemble/request_data.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        return None
+
+    @property
+    def parameters(self):
+        return ParametersDict("parameters", "key", self)
+
+    @property
+    def responses(self):
+        return ParametersDict("responses", "name", self)
```
Comment (ManInFez) [inline ertviz/ertapi/ensemble/request_data.py:76]: are all these parametersdict? odd naming when have a strong notion of what parameters are in ERT

File: ertviz/ertapi/ensemble/request_data.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        self._parent = parent
+        self._key = key
+        self._index = -1
+
+    def __iter__(self):
+        return self
+
+    def _get_gen_class(self):
+        return {
+            "parameters": ertviz.ertapi.ensemble.Parameter,
```
Comment (ManInFez) [inline ertviz/ertapi/ensemble/request_data.py:18]: Not necessary to create this dict every time

File: ertviz/ertapi/ensemble/request_data.py (hunk 2 above)
Comment (xjules) [reply ertviz/ertapi/ensemble/request_data.py:76]: Agree, this should be change to something better. Essentially, it unifies access to objects:
```python
ensemble.responses.name #gets all names of responses, without creating responses that is
ensembles.responses["POLY_RES"] #gets a single response object
```

File: ertviz/ertapi/ensemble/observation.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def _convertdate(dstring):
+    return datetime.strptime(dstring, "%Y-%m-%d %H:%M:%S")
+
+
+def indexes_to_axis(indexes):
+    if indexes and ":" in indexes[0]:
+        return list(map(_convertdate, indexes))
+    return list(map(int, indexes))
```
[the remark this answers is not in the document]
Comment (xjules) [reply ertviz/ertapi/ensemble/observation.py:12]: I must confess that I just copied from somewhere else too :)

**Comment by xjules** [comment]
I implemented a basic statistics plot for realizations, feel free to comment
 
image

File: ertviz/ertapi/ensemble/request_data.py (hunk 3 above)
Comment (xjules) [reply ertviz/ertapi/ensemble/request_data.py:18]: True, result of a quick refactoring yesterday :)

**Comment by oysteoh** [comment]
i think that is really good! get the views up and running, then we have something to discuss 👍 👍  

is it possible having this view in its own PR without your changes to the models? so that we can adjust those afterwards if we want? now everything is squashed together into this big PR with a lot of changes..... 🤔

**Comment by xjules** [comment]
Decided no to use `ertapi_model`, so closing this PR.

**Pull request closed without merging**